utils/verify_intel_gaudi/roles/verify_intel_gaudi/tasks/hlqual_validation.yml

```diff
@@ -0,0 +1,279 @@
+# Copyright 2024 Intel Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+
+- name: Setting python version used for the test runs
+  ansible.builtin.set_fact:
+    pver: "{{ '.'.join(ansible_python_version.split('.')[0:2]) }}"
+
+- name: Recursively change ownership of habana_logs directory
+  ansible.builtin.file:
+    path: "{{ verify_intel_gaudi_habana_tests['habana_logs'] }}"
+    state: directory
+    recurse: true
+    mode: "{{ verify_intel_gaudi_habana_tests['habana_logs_permission'] }}"
+
+- name: Run hl_qual hardware sanity check test
+  environment:
+    __python_cmd: "python{{ pver }}"
+    LOG_LEVEL_ALL: "{{ verify_intel_gaudi_habana_tests['log_level_all'] }}"
+    ENABLE_CONSOLE: "{{ verify_intel_gaudi_habana_tests['enable_console'] }}"
+    HABANA_LOGS: "{{ verify_intel_gaudi_habana_tests['habana_logs'] }}"
+    GC_KERNEL_PATH: "{{ verify_intel_gaudi_habana_tests['gc_kernel_path'] }}"
+    HABANA_SCAL_BIN_PATH: "{{ verify_intel_gaudi_habana_tests['habana_scal_bin_path'] }}"
+    HABANA_PLUGINS_LIB_PATH: "{{ verify_intel_gaudi_habana_tests['habana_plugins_lib_path'] }}"
+    DATA_LOADER_AEON_LIB_PATH: "{{ verify_intel_gaudi_habana_tests['data_loader_aeon_lib_path'] }}"
+  ansible.builtin.shell: |
+    set -o pipefail
+    ./hl_qual -gaudi2 -c all -rmod parallel -f2 -l extreme -t 60 -dis_mon
+  args:
+    executable: /bin/bash
+    chdir: /opt/habanalabs/qual/gaudi2/bin
+  register: sanity_test_result
+  failed_when: "'FAILED' in sanity_test_result.stdout"
+  changed_when: true
+
+- name: Run other tests
+  when: not verify_intel_gaudi_with_sanity_test_only
```

Comment:
we can add when condition in main.yml directly instead of individual task

Reply:
this only applies to hlqual